crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -45,21 +45,27 @@ def load!
           end
         end
 
-        etc_repos.each do |platform, repos|
+        etc_repos.each do |platform, arches|
           if @all_repos.key? platform
-            repos.each do |id, repo|
-              # for repos that exist in our hard-coded file, we only allow
-              # overwriting a subset of attributes
-              if @all_repos[platform].key? id
-                %w(url ask_on_error).each do |key|
-                  @all_repos[platform][id][key] = repo[key] if repo.key? key
+            arches.each do |arch, repos|
+              if @all_repos[platform].key? arch
+                repos.each do |id, repo|
+                  # for repos that exist in our hard-coded file, we only allow
+                  # overwriting a subset of attributes
+                  if @all_repos[platform][arch].key? id
+                    %w(url ask_on_error).each do |key|
+                      @all_repos[platform][arch][id][key] = repo[key] if repo.key? key
```

Comment:
Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)